recipes/perl/rbac_roles/rbac_user_migration.pl

```diff
@@ -0,0 +1,592 @@
+{
+	"data": {
+		"type": "accessControlRole",
+		"attributes": {
+			"name": "Security Administrator",
+			"description": "Security administrator with privileges for performing actions including access control, certificates, security settings etc."
+		}
+	}
+}
+EOL
+		$response = $http->post($access_control_roles_url, 'Content-Type' => $media_type, Content => $security_role_admin_body);
+		unless ($response->is_success) {
+			if ($debug) {
+				print Dumper($response);
+			}
+
+			die "Failed to create security administrator role!\n" ;
+		}
+		my $response_content = decode_json($response->content);
+		$new_security_admin_role_id = $response_content->{data}{id};
+		print "New Security Administrator role created with ID: ", $new_security_admin_role_id, "\n";
+	}
+}
+
+# Create access definitions for new Security Administrator role
+if ($do_security =~ /y/i) {
+	for my $managed_object (keys %security_role_operation_map) {
+		my $access_definitions_url = "https://$hostname/netbackup/access-control/managed-objects/$managed_object/access-definitions";
+		my $operations = $security_role_operation_map{$managed_object};
+		my $access_definition_object = encode_json(generate_access_definition_object($new_security_admin_role_id, $managed_object, $operations));
+		$response = $http->post($access_definitions_url, 'Content-Type' => $media_type, Content => $access_definition_object);
+		unless ($response->is_success) {
+			if ($debug) {
+				print Dumper($response);
+			}
+			die "Failed to create access definition for security administrator role!\n";
+		}
+	}
+}
+
+# Add security user principals to new Security Administrator role
+if (@security_user_principals && $do_security =~ /y/i) {
+	my $user_security_role_url = "https://$hostname/netbackup/access-control/roles/$new_security_admin_role_id/relationships/user-principals";
+	my $security_users = encode_json(generate_user_principal_resource_identifier_objects(@security_user_principals));
+	$response = $http->post($user_security_role_url, 'Content-Type' => $media_type, Content => $security_users);
+	unless ($response->is_success) {
+		if ($debug) {
+			print Dumper($response);
+		}
+		die "Failed to add security admin users!\n" ;
+	}
+}
+
+# Add security group principals to new Security Administrator role
+if (@security_group_principals && $do_security =~ /y/i) {
+	my $group_security_role_url = "https://$hostname/netbackup/access-control/roles/$new_security_admin_role_id/relationships/group-principals";
+	my $security_groups = encode_json(generate_group_principal_resource_identifier_objects(@security_group_principals));
+	$response = $http->post($group_security_role_url, 'Content-Type' => $media_type, Content => $security_groups);
+	unless ($response->is_success) {
+		if ($debug) {
+			print Dumper($response);
+		}
+		die "Failed to add security admin groups!\n" ;
+	}
+}
+
+
+sub extract_role_ids {
+	my ($response) = @_;
+	my $content = decode_json($response->content);
+	my @data = $content->{data};
```

Comment:
This creates a one-item list holding the list ref. Probably not what's intended.

```suggestion
	my @data = @{ $content->{data} };
```

The array isn't used, though. I think it was meant to be used in the loop two lines below.